audit/src/main/java/org/wildfly/security/audit/SyslogAuditEndpoint.java

```diff
@@ -65,7 +75,23 @@ public void accept(EventPriority priority, String message) throws IOException {
         synchronized(this) {
             if (!accepting) return;
 
-            syslogHandler.doPublish(new ExtLogRecord(toLevel(priority), message, SyslogAuditEndpoint.class.getName()));
+            while(true) {
+                try {
+                    tryPublish(priority, message);
+                    break;
+                } catch (IOException e) {
+                    // TcpOutputStream has its' own reconnect handler, so just throw the error
+                    if (protocol != Protocol.UDP) {
+                        throw e;
+                    }
+                    if (currentReconnectAttempts == maxReconnectAttempts) {
+                        throw audit.syslogMaximumReconnectAttemptsReached(currentReconnectAttempts);
+                    } else if (maxReconnectAttempts != -1) {
+                        // Reconnect attempts are less than max so eat the error
+                        currentReconnectAttempts++;
```

Comment:
I think in the case and the one above, it would be good to still log the error message and indicate that it will attempt to reconnect.

Reply:
Added a log method here